go-controller/pkg/ovn/obj_retry.go

```diff
@@ -711,6 +797,103 @@ func (oc *Controller) updateResource(objectsToRetry *retryObjs, oldObj, newObj i
 			extraParameters.portGroupIngressDenyName,
 			extraParameters.portGroupEgressDenyName,
 			newObj)
+	case factory.EgressIPType:
+		oldEIP := oldObj.(*egressipv1.EgressIP)
+		newEIP := newObj.(*egressipv1.EgressIP)
+		if err := oc.reconcileEgressIP(oldEIP, newEIP); err != nil {
+			return fmt.Errorf("unable to update EgressIP: %s, err: %v", newEIP.Name, err)
+		}
+		return nil
+	case factory.EgressIPNamespaceType:
+		oldNamespace := oldObj.(*kapi.Namespace)
+		newNamespace := newObj.(*kapi.Namespace)
+		if err := oc.reconcileEgressIPNamespace(oldNamespace, newNamespace); err != nil {
+			return fmt.Errorf("unable to update egress IP matching namespace: %s, err: %v", newNamespace.Name, err)
+		}
+		return nil
+	case factory.EgressIPPodType:
+		oldPod := oldObj.(*kapi.Pod)
+		newPod := newObj.(*kapi.Pod)
+		if util.PodCompleted(newPod) {
```

Comment:
Here I see the same pattern used for pods and dynamic pod watchers: the pod is in terminal/completed state and we want be done with the update by just removing the corresponding resource.

I would add `factory.EgressIPPodType` to `isObjectInTerminalState` as in:
```
func (oc *Controller) isObjectInTerminalState(objType reflect.Type, obj interface{}) bool {
	switch objType {
	case factory.PodType,
		factory.PeerPodSelectorType,
		factory.PeerPodForNamespaceAndPodSelectorType,
		factory.LocalPodSelectorType,
		factory.EgressIPPodType:
		pod := obj.(*kapi.Pod)
		return util.PodCompleted(pod)

	default:
		return false
	}
}
```

Then, if the pod is completed, `processObjectInTerminalState` will take care of removing the corresponding resource by calling `deleteResource`.

In this part of `updateResource`, I would just call `oc.reconcileEgressIPPod(oldPod, newPod)`

Reply:
`processObjectInTerminalState` as is doesn't what egresspod needs ?

Reply:
it does if we tell it to care about this resource type, so the change I propose is to:
- modify `isObjectInTerminalState` as above
- here remove the whole part that checks for a completed pod, since `isObjectInTerminalState` and `processObjectInTerminalState` will take care of it:

```
	case factory.EgressIPPodType:
		oldPod := oldObj.(*kapi.Pod)
		newPod := newObj.(*kapi.Pod)
		return oc.reconcileEgressIPPod(oldPod, newPod)
```